pkg/importer/language.go

```diff
@@ -62,22 +65,93 @@ func (e *DiagnosticsError) String() string {
 	return e.Error()
 }
 
-func removeDuplicatePathedValues(pathedValues []PathedLiteralValue) []PathedLiteralValue {
+func removeDuplicateIdentityPathedValues(pathedValues []PathedLiteralValue) []PathedLiteralValue {
 	uniqueValues := make([]PathedLiteralValue, 0)
 	occurrences := make(map[string]int)
 	for _, pathedValue := range pathedValues {
-		occurrences[pathedValue.Value]++
+		if pathedValue.Identity {
+			occurrences[pathedValue.Value]++
+		}
 	}
 
 	for _, pathedValue := range pathedValues {
-		if occurrences[pathedValue.Value] > 1 {
+		if pathedValue.Identity && occurrences[pathedValue.Value] > 1 {
```

Comment:
Why is this just being checked for Identity values? Isn't this a problem for any value? If two resources both have an output "hello" and another resource has an input "hello" you can't decided which of the two resources to draw from.

Reply:
So the reason for this because if you have for example these pathed values computed
```
resourceGroup.id="group123"
resourceA.resourceGroup = "group123"
resourceB.resourceGroup = "group123"
```
Then I wanted to maintain the path `resourceGroup.id` even if that value is duplicated in paths `resourceA.resourceGroup` and `resourceB.resourceGroup`. 

However, I think when building the pathed values, I should only be using `state.Outputs` instead of `state.Inputs` and only replace literal inputs with paths of resource outputs. Even then, `resourceA.resourceGroup` and `resourceB.resourceGroup` will occur both on `state.Inputs` and `state.Outputs` AFAIK

Would be great if I can actually test this with the CLI but getting weird behaviour with pulumi import

Reply:
Maybe I can actually do a second pass where we remove duplicate paths without taking identities into account and then merge non-duped identities and non-duped values 🤔

Reply:
> Maybe I can actually do a second pass where we remove duplicate paths without taking identities into account and then merge non-duped identities and non-duped values 🤔

Implemented ☝️ 

Also found an edge case of "overguessing" that could result in a circular reference:
```ts
const bucket = new aws.s3.Bucket("my-bucket", {
    website: {
        indexDocument: "index.html",
    },
});

const bucketObject = new aws.s3.BucketObject("index.html", {
    bucket: bucket.id
});
```

Implemented a fallback such that if we encounter this error, we retry generating the program without guessing references.

Reply:
This has been resolved with the `ancestorTypes` feature where you can specify that `BucketObject` can guess values from outputs of `Bucket` but not the other way around ✅